upup/pkg/fi/utils/hash.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"strings"
+)
+
+// HashString Takes String and returns a sha1 hash represented as a string
+func HashString(s *string) (string, error) {
```

Comment:
We probably shouldn't take a string pointer - it's a ref type anyway, and we can't pass nil